filter_plugins/filters.py

```diff
@@ -16,11 +17,14 @@ def grub2_hash(password, salt=None, iterations=10000):
         r = SystemRandom()
         salt = ''.join([r.choice(string.ascii_letters + string.digits) for _ in range(64)])
 
+    # Make sure we have passlib and the correct version
     if not HAS_PASSLIB:
         raise errors.AnsibleFilterError('grub2_hash requires the passlib python module to generate password hashes')
-    else:
-        encrypted = passlib.hash.grub_pbkdf2_sha512.hash(password, salt=salt, rounds=iterations)
 
+    if PASSLIB_VERSION < 1.7:
```

Comment:
grub_pbkdf2_sha512 was added in passlib-1.4: http://passlib.readthedocs.io/en/1.7.1/history/ancient.html

Here's the original commit: https://bitbucket.org/ecollins/passlib/commits/2650a3ba3d44e13f10733ebe9fe590f70c53e022